odpf/guardian/guardian.proto

```diff
@@ -0,0 +1,375 @@
+        option (google.api.http) = {
+            post: "/appeals"
+            body: "*"
+        };
+    }
+
+    rpc ListApprovals(ListApprovalsRequest) returns (ListApprovalsResponse) {
+        option (google.api.http) = {
+            get: "/appeals/approvals"
+        };
+    }
+
+    rpc UpdateApproval(UpdateApprovalRequest) returns (UpdateApprovalResponse) {
+        option (google.api.http) = {
+            post: "/appeals/{id}/approvals/{approval_name}"
+            body: "*"
+        };
+    }
+}
+
+message ListProvidersRequest {}
+
+message ListProvidersResponse {
+    repeated Provider providers = 1;
+}
+
+message CreateProviderRequest {
+    Provider provider = 1;
+}
+
+message CreateProviderResponse {
+    Provider provider = 1;
+}
+
+message UpdateProviderRequest {
+    uint32 id = 1;
+    Provider provider = 2;
+}
+
+message UpdateProviderResponse {
+    Provider provider = 1;
+}
+
+message ListPoliciesRequest {}
+
+message ListPoliciesResponse {
+    repeated Policy policies = 1;
+}
+
+message CreatePolicyRequest {
+    Policy policy = 1;
+}
+
+message CreatePolicyResponse {
+    Policy policy = 1;
+}
+
+message UpdatePolicyRequest {
+    string id = 1;
+    Policy policy = 2;
+}
+
+message UpdatePolicyResponse {
+    Policy policy = 1;
+}
+
+message ListResourcesRequest {}
+
+message ListResourcesResponse {
+    repeated Resource resources = 1;
+}
+
+message UpdateResourceRequest {
+    uint32 id = 1;
+    Resource resource = 2;
+}
+
+message ListAppealsRequest {
+    string user = 1;
+}
+
+message ListAppealsResponse {
+    repeated Appeal appeals = 1;
+}
+
+message GetAppealRequest {
+    uint32 id = 1;
+}
+
+message GetAppealResponse {
+    Appeal appeal = 1;
+}
+
+message CancelAppealRequest {
+    uint32 id = 1;
+}
+
+message CancelAppealResponse {
+    Appeal appeal = 1;
+}
+
+message RevokeAppealRequest {
+    uint32 id = 1;
+}
+
+message RevokeAppealResponse {
+    Appeal appeal = 1;
+}
+
+message CreateAppealRequest {
+    string user = 1;
+
+    message Resource {
+        uint32 id = 1;
+        string role = 2;
+        google.protobuf.Struct options = 3;
+    }
+    repeated Resource resources = 2;
+}
+
+message CreateAppealResponse {
+    repeated Appeal appeals = 1;
+}
+
+message ListApprovalsRequest {
+    string user = 1;
+}
+
+message ListApprovalsResponse {
+    repeated Approval approvals = 1;
+}
+
+message UpdateApprovalRequest {
+    uint32 id = 1;
+    string approval_name = 2;
+
+    message Action {
+        string action = 1;
+    }
+    Action action = 3;
+}
+
+message UpdateApprovalResponse {
+    Appeal appeal = 1;
+}
+
+// Provider contains information about external data provider such as BigQuery, Metabase, etc., credentials, policy, and allowed roles 
+message Provider {
+    uint32 id = 1;
+    string type = 2;
+    string urn = 3;
+
+    message ProviderConfig {
+        string type = 1;
+        string urn = 2;
+        google.protobuf.Struct labels = 3;
```

Comment:
I see labels as `key,value` pairs of string, any reason why we chose Struct for this? We should limit the usage of `Struct` to only places where we know for sure we can't use concrete types I think as it adds inconvenience while parsing it.

Reply:
The type we have for labels in guardian right now is `map[string]interface{}`, using Struct to fulfill that.

Reply:
I think this is better moved to [string][string] labels should not be In any other type.

Reply:
changing the proto type for labels to `map<string, string>`. Will change the labels type in guardian to `map[string]string` as well